sdk/translation/azure-ai-translation-text/customization/src/main/java/TextTranslationClientBuilderCustomization.java

```diff
@@ -0,0 +1,54 @@
+import com.azure.autorest.customization.ClassCustomization;
+import com.azure.autorest.customization.Customization;
+import com.azure.autorest.customization.LibraryCustomization;
+import com.azure.autorest.customization.PackageCustomization;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import org.slf4j.Logger;
+
+/**
+ * This class contains the customization code to customize the TypeSpec generated code for Text Translation Client Builder.
+ */
+public class TextTranslationClientBuilderCustomization extends Customization {
+
+    @Override
+    public void customize(LibraryCustomization customization, Logger logger) {
+
+        logger.info("Customizing the TextTranslationClientBuilder class");
+        ClassCustomization classCustomizationForTextTranslationClientBuilder = customization.getClass("com.azure.ai.translation.text", "TextTranslationClientBuilder");
+
+        addAzureKeyCredentialTrait(classCustomizationForTextTranslationClientBuilder);
+        addTokenCredentialTrait(classCustomizationForTextTranslationClientBuilder);
+    }
+
+    private void addAzureKeyCredentialTrait(ClassCustomization classCustomization) {
+        // add AzureKeyCredentialTrait imports
+        classCustomization.addImports("com.azure.core.client.traits.AzureKeyCredentialTrait");
+
+        // add KeyCredentialTrait<> as implemented type
+        classCustomization.customizeAst(compilationUnit -> {
+            ClassOrInterfaceDeclaration jobRouterAdministrationClientBuilderClass = compilationUnit.getClassByName(classCustomization.getClassName()).get().asClassOrInterfaceDeclaration();
+            NodeList<ClassOrInterfaceType> implementedTypes = jobRouterAdministrationClientBuilderClass.getImplementedTypes();
+            boolean hasKeyCredentialTrait = implementedTypes.stream().filter(implementedType -> implementedType.getNameAsString().equals("AzureKeyCredentialTrait")).findFirst().isPresent();
```

Comment:
new lib should use `KeyCredentiaTrait`.

Reply:
And why the authentication is not defined in typespec?
https://typespec.io/docs/libraries/http/authentication

Reply:
Our authentication for the service is not following current guidelines for auth and have many special cases that are not supported by the TypeSpec. We are doing same for all languages - adding methods that are not available.

Reply:
OK. But consider changing it to `KeyCredential` (without Azure).

Reply:
OK, I switched to `KeyCredential`.